src/lib.rs

```diff
@@ -133,22 +133,20 @@ fn no_default_default_rule() -> Rule {
                           )
                       ) @c
                     )"#,
```

Comment:
This seems like it could potentially take the raw query without being wrapped in a string (at the expense of the parsing being more complicated, I guess).

Reply:
Intriguing ya I don't think it's guaranteed that there's an "outermost pair of enclosing parentheses" (I don't totally understand that though) in which case I think ya parsing could be trickier

Oh also there are places where this can be a (more general) Rust expression rather than a string literal, eg I used `format!()` somewhere to "generate" the query string as a function of some "rule state"